programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/VFSNodeScratchSpaceImpl.java

```diff
@@ -117,17 +118,21 @@ private AppScratchSpaceImpl(final long appId) throws FileSystemException {
 
         public void close() throws FileSystemException {
             logger.debug("Closing application scratch space");
+
             try {
+                final int filesNumber = spaceFile.delete(Selectors.SELECT_ALL);
+                logger.debug("Deleted " + filesNumber + " files in scratch application directory");
+            } catch (org.apache.commons.vfs2.FileSystemException e) {
+                logger.warn("Could not delete " + spaceFile, e);
+            } finally {
+                // just a hint
                 try {
-                    final int filesNumber = spaceFile.delete(Selectors.SELECT_ALL);
-                    logger.debug("Deleted " + filesNumber + " files in scratch application directory");
-                } finally {
-                    // just a hint
                     spaceFile.close();
```

Comment:
It is still part of a finally block if you lock at the full change. I have change the imbrication in order to catch a file deletion that fails only.